fuel-core/src/state/rocks_db.rs

```diff
@@ -171,7 +193,16 @@ impl KeyValueStore for RocksDb {
         let iter = self
             .db
             .iterator_cf_opt(&self.cf(column), opts, iter_mode)
-            .map(|(key, value)| (key.to_vec(), value.to_vec()));
+            .map(|(key, value)| {
+                #[cfg(feature = "prometheus")]
+                {
+                    DATABASE_METRICS.read_meter.inc();
+                    DATABASE_METRICS
+                        .bytes_read_meter
+                        .inc_by(value.to_vec().len() as u64);
```

Comment:
1. We should convert value to vec before logging the metrics so we don't need to double copy the bytes just to check the length.
2. the key byte length should also be included

Reply:
Had to fix merge conflicts so corrected this anyway

Reply:
ok we should still file a tech debt to make sure the key length is also included in the read bytes.

We also need to instrument the batch_writes

Reply:
Filed in #416